vital/system.py

```diff
@@ -40,6 +40,22 @@ def __init__(
         # Also save the classpath of the system to be able to load a checkpoint w/o knowing it's type beforehand
         self.save_hyperparameters({"task": {"_target_": f"{self.__class__.__module__}.{self.__class__.__name__}"}})
 
+    @classmethod
+    def load_from_checkpoint(cls, *args, **kwargs):  # noqa: D102
+        if ABC in cls.__bases__:
+            # We use this method to determine if the class is abstract because Lightning does not use the
+            # `@abstractmethod` decorator to mark abstract methods (e.g. `train_step`), rather relying on runtime
+            # warnings. Because of this, we cannot rely on Python's canonical function for detecting abstract classes,
+            # `inspect.isabstract`, which expects abstract classes to have explicitly defined abstract methods.
+            raise NotImplementedError(
+                f"Class '{cls.__name__}' does not support being loaded from a checkpoint because it is an ABC. Either "
+                f"call `load_from_checkpoint` on the specific class of the system you want to load, or use the utility "
+                f"function `load_from_checkpoint` provided in `vital.utils.saving` that will automatically detect the "
+                f"system class when loading the checkpoint."
+            )
```

Comment:
What about resolving the class automatically?

```suggestion
           return vital.utils.saving.load_from_checkpoint(checkpoint_path)
```

I tried something like this and it worked.

Reply:
Ça prendrait juste un commentaire différent, qui expliquerait la situation. On pourrait peut-être juste dire que `VitalSystem.load_from_checkpoint` est un alias pour `vital.utils.saving.load_from_checkpoint` ?

Reply:
Personnellement, je ne suis pas encore convaincu de l'idée d'utiliser `vital.utils.saving.load_from_checkpoint` à l'intérieur de la méthode. Je comprends qu'initialement c'est plus intuitif d'utiliser la méthode que de découvrir la fonction utilitaire. Toutefois, la fonction se veut être une version "optimisée" pour un cas normal où l'on veut loader un modèle, mais elle cache le reste de l'API de la méthode (p.ex. `map_location`, `hparams_file`, etc.).

À cause de ça, je dirais que son rôle (loading très simple d'un modèle pour qu'il soit prêt pour l'inférence) commence à être suffisamment différent de celui de la méthode (API complet pour loader un modèle) pour justifier qu'elle reste une fonction indépendante, et que la méthode reste accessible pour les cas où on veut accéder à ou modifier le comportement natif à Lightning.

Reply:
Quand je dis que je propose que la méthode `VitalSystem.load_from_checkpoint` soit un alias de la fonction `vital.utils.saving.load_from_checkpoint`, ce n'est pas tout à fait exact; pour être plus précis, la méthode agirait comme un alias si on l'appelle directement (`cls == VitalSystem`). Or, si on l'appelle sur une classe "instanciable" (e.g. `cls == SegmentationTask`), on a accès à travers elle à l'API de Lightning. Donc ce n'est pas contraignant, on a accès aux deux approches.

Cependant, c'est probablement trop mélangeant. La fonction est un wrapper autour de la méthode, mais si on prend ma proposition, la méthode serait dans un certain cas un alias d'un wrapper d'elle-même... La méthode agit différemment selon l'identité de `cls`.

Finalement je suis d'accord de garder le message d'erreur. Cependant, si la méthode et la fonction sont bien différentes, est-ce qu'on devrait renommer la fonction pour mettre l'accent sur cette différence? Malheureusement je n'ai pas de nom concis à proposer, qui n'inclurait pas plusieurs verbes (e.g. `detect_class_and_load_from_checkpoint`). Si rien ne te vient, laissons comme ça.

Reply:
Je n'ai pas de bonne idées pour un nom qui différencierait bien les deux sans être trop verbose non plus. Donc je serais de l'opinion de garder le nom actuel dans le doute :wink:

Reply:
Ça me va